cmd/vela-worker/operate.go

```diff
@@ -160,13 +167,18 @@ func (w *Worker) operate(ctx context.Context) error {
 					// (do not pass the context to avoid errors in one
 					// executor+build inadvertently canceling other builds)
 					//nolint:contextcheck // ignore passing context
-					err = w.exec(id)
+					err = w.exec(id, registryWorker)
 					if err != nil {
 						// log the error received from the executor
 						//
 						// https://pkg.go.dev/github.com/sirupsen/logrus?tab=doc#Errorf
 						logrus.Errorf("failing worker executor: %v", err)
-
+						registryWorker.SetStatus(constants.WorkerStatusError)
+						_, res, ers := w.VelaClient.Worker.Update(registryWorker.GetHostname(), registryWorker)
```

Comment:
i think we also need to check to make sure res is not nil somewhere, since we grab the statuscode from it for the error